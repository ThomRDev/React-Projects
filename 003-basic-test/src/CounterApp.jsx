import PropTypes from "prop-types"

import { useState } from "react"

const CounterApp = ({ initValue }) => {

    const [counter, setCounter] = useState(initValue)

    console.log("renderizo");
    const handleAdd = (event) => {
        // setCounter(counter + 1)
        setCounter(oldValue => oldValue + 1)
    }
    const handleSubtract = (event) => {
        if (counter <= 0) return
        // setCounter(counter - 1)
        setCounter(oldValue => oldValue - 1)
    }

    const handleReset = (event) => {
        setCounter(initValue)
    }
    return (
        <div className="App">
            <header className="App-header">
                <div className="value">{counter}</div>
                <div className='actions'>
                    <button onClick={handleAdd}>+1</button>
                    <button onClick={handleReset}>Reset</button>
                    <button onClick={handleSubtract}>-1</button>
                </div>
            </header>
        </div>
    );
}

CounterApp.propTypes = {
    initValue: PropTypes.number
}
CounterApp.defaultProps = {
    initValue: 1
}

export default CounterApp;