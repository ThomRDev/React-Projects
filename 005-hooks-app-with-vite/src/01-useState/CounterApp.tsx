import { useState } from "react"

export const CounterApp = () => {
  const [counter, setCounter] = useState({
    counter1:10,
    counter2:20,
    counter3:30,
  })
  return (
    <>
      <h2>Counter : { JSON.stringify(counter) }</h2>
      <hr />
      <button className="btn btn-primary" onClick={()=>setCounter({
        ...counter,
        counter1 : counter.counter1 + 1
      })} >+1 Counter1</button>
    </>
  )
}
