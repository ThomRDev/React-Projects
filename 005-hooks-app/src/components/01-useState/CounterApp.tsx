import React, { useState } from 'react'

const CounterApp = () => {
    const [{ counter1,counter2 }, setCounter] = useState({
        counter1 : 0,
        counter2 : 0
    })

    const handleActionClickCounter1 = (event:React.MouseEvent<HTMLButtonElement>) =>{
        event.preventDefault()
        const {action} = event.currentTarget.dataset
        if(action === "decrement"){
            setCounter(oldValue=>({
                ...oldValue,
                counter1 : counter1 - 1 
            }))
        }
    }
    const handleActionClickCounter2 = (event:React.MouseEvent<HTMLButtonElement>) =>{
        event.preventDefault()
        const {action} = event.currentTarget.dataset
        if(action === "decrement"){
            setCounter(oldValue=>({
                ...oldValue,
                counter2 : counter2 - 1 
            }))
        }
    }
    return (
        <div className='d-flex align-items-center gap-5'>  
            <section>
                <h2>Counter-1 : { counter1 }</h2>
                <div className="d-flex align-items-center gap-2">
                    <button onClick={ handleActionClickCounter1 } className="btn btn-primary d-grid fs-5" style={{placeItems:"center"}} data-action="increment" data-item="counter1">+1</button>
                    <button onClick={ handleActionClickCounter1 } className="btn btn-danger d-grid fs-5" style={{placeItems:"center"}} data-action="decrement" data-item="counter1">-1</button>
                    <button onClick={ handleActionClickCounter1 } className="btn btn-secondary d-grid fs-5" style={{placeItems:"center"}} data-action="reset" data-item="counter1">Reset</button>
                </div>
            </section>
            <section>
                <h2>Counter-2 : { counter2 }</h2>
                <div className="d-flex align-items-center gap-2">
                    <button onClick={ handleActionClickCounter2 } className="btn btn-primary d-grid fs-5" style={{placeItems:"center"}} data-action="increment" data-item="counter2">+1</button>
                    <button onClick={ handleActionClickCounter2 } className="btn btn-danger d-grid fs-5" style={{placeItems:"center"}} data-action="decrement" data-item="counter2">-1</button>
                    <button onClick={ handleActionClickCounter2 } className="btn btn-secondary d-grid fs-5" style={{placeItems:"center"}} data-action="reset" data-item="counter2">Reset</button>
                </div>
            </section>
            
        </div>
    )
}

export default CounterApp