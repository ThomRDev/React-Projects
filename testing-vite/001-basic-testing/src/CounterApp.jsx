import PropTypes from "prop-types"

import { useState } from "react"

export const CounterApp = ({ initValue }) => {

    const [counter, setCounter] = useState(initValue)

    const handleAdd = (event) => {
        setCounter(oldValue => oldValue + 1)
    }
    const handleSubtract = (event) => {
        if (counter <= 0) return
        setCounter(oldValue => oldValue - 1)
    }

    const handleReset = (event) => {
        setCounter(initValue)
    }
    return (
        <div className="App">
            <header className="App-header">
                <div className="value">{counter}</div>
                <div className='actions'>
                    <button onClick={handleAdd}>+1</button>
                    <button onClick={handleReset}>Reset</button>
                    <button onClick={handleSubtract}>-1</button>
                </div>
            </header>
        </div>
    );
}

CounterApp.propTypes = {
    initValue: PropTypes.number
}
CounterApp.defaultProps = {
    initValue: 1
}
