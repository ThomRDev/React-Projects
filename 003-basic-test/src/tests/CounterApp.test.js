import { shallow } from "enzyme"
import CounterApp from "../CounterApp";

describe(`Tests in ${__filename} file`, () => {
    test('Must return component <CounterApp /> with default value', () => {
        const wrapper = shallow(<CounterApp />)
        expect(wrapper).toMatchSnapshot()
        const initValue = wrapper.find(".value").text().trim()
        expect(+initValue).toBe(1)
    })
    test('Must return component <CounterApp /> with value = 5', () => {
        const wrapper = shallow(<CounterApp initValue={5} />)
        expect(wrapper).toMatchSnapshot()

        const initValue = wrapper.find(".value").text().trim()
        expect(+initValue).toBe(5)
    })
    
    test('click btn +1 => result 2', () => { 
        const wrapper = shallow(<CounterApp />)

        // el primer boton - que es el de +1
        const btn = wrapper.find("button").at(0)
        btn.simulate("click")

        expect(+wrapper.find(".value").text().trim()).toBe(2)
    })
    test('click btn -1 => result 5 with initValue 6', () => { 
        const wrapper = shallow(<CounterApp initValue={6} />)

        const btn = wrapper.find("button").at(2)
          // console.log(wrapper.find('button').at(2).html())
        btn.simulate("click")

        expect(+wrapper.find(".value").text().trim()).toBe(5)
    })
    test('click reset => result 100 with initValue 100', () => { 
        const wrapper = shallow(<CounterApp initValue={100} />)

        const btn = wrapper.find("button").at(1)
        btn.simulate("click")

        expect(+wrapper.find(".value").text().trim()).toBe(100)
    })
});

describe(`Tests in ${__filename} file - reuse`, () => {
    
    // se inicializa para poder tener el autoayuda, pero deberia ir = null
    let wrapper = shallow(<CounterApp />)
    let value = null

    // se ejecutara antes de cada rest
    beforeEach(()=>{
        value = 5
        wrapper = shallow(<CounterApp initValue={value} />);
    })
    test('Must return component <CounterApp /> with value', () => {
        expect(wrapper).toMatchSnapshot()
        const initValue = wrapper.find(".value").text().trim()
        expect(+initValue).toBe(value)
        
    })
    test("clicks verify",()=>{
        // +5 => 5 +5 = 10
        wrapper.find("button").at(0).simulate("click")
        wrapper.find("button").at(0).simulate("click")
        wrapper.find("button").at(0).simulate("click")
        wrapper.find("button").at(0).simulate("click")
        wrapper.find("button").at(0).simulate("click")
        
        expect(+wrapper.find(".value").text().trim()).toBe(10)
        
        // -2
        // =>10 -2 =>8
        wrapper.find("button").at(2).simulate("click")
        wrapper.find("button").at(2).simulate("click")
        expect(+wrapper.find(".value").text().trim()).toBe(8)
        
        // reset
        wrapper.find("button").at(1).simulate("click")
        expect(+wrapper.find(".value").text().trim()).toBe(5)
    })
});
