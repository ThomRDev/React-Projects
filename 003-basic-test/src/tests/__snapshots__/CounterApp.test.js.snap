// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in C:\\Users\\ThomMR\\Desktop\\Learning-React-2022\\003-basic-test\\src\\tests\\CounterApp.test.js file - reuse Must return component <CounterApp /> with value 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <div
      className="value"
    >
      5
    </div>
    <div
      className="actions"
    >
      <button
        onClick={[Function]}
      >
        +1
      </button>
      <button
        onClick={[Function]}
      >
        Reset
      </button>
      <button
        onClick={[Function]}
      >
        -1
      </button>
    </div>
  </header>
</div>
`;

exports[`Tests in C:\\Users\\ThomMR\\Desktop\\Learning-React-2022\\003-basic-test\\src\\tests\\CounterApp.test.js file Must return component <CounterApp /> with default value 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <div
      className="value"
    >
      1
    </div>
    <div
      className="actions"
    >
      <button
        onClick={[Function]}
      >
        +1
      </button>
      <button
        onClick={[Function]}
      >
        Reset
      </button>
      <button
        onClick={[Function]}
      >
        -1
      </button>
    </div>
  </header>
</div>
`;

exports[`Tests in C:\\Users\\ThomMR\\Desktop\\Learning-React-2022\\003-basic-test\\src\\tests\\CounterApp.test.js file Must return component <CounterApp /> with value = 5 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <div
      className="value"
    >
      5
    </div>
    <div
      className="actions"
    >
      <button
        onClick={[Function]}
      >
        +1
      </button>
      <button
        onClick={[Function]}
      >
        Reset
      </button>
      <button
        onClick={[Function]}
      >
        -1
      </button>
    </div>
  </header>
</div>
`;
